include/deal.II/base/utilities.h

```diff
@@ -946,40 +961,48 @@ namespace Utilities
       &&
       sizeof(T)<256)
       {
-        std::vector<char> buffer (sizeof(T));
-        std::memcpy (buffer.data(), &object, sizeof(T));
-        return buffer;
+        const size_t previous_size = dest_buffer.size();
+        dest_buffer.resize (previous_size + sizeof(T));
+
+        std::memcpy (dest_buffer.data() + previous_size, &object, sizeof(T));
       }
     else
       {
-        // set up a buffer and then use it as the target of a compressing
+        // use buffer as the target of a compressing
         // stream into which we serialize the current object
-        std::vector<char> buffer;
-        {
```

Comment:
if you keep the brackets the diff will be smaller and more apparent what changes.

Also, please run indentation script.

Reply:
@marcfehling please keep the bracket so that the diff is smaller

Reply:
Totally forgot about those! I"ll update that on the next commit.
But the result with the double curly brackets looks unsatisfying imho.

Reply:
> the result with the double curly brackets looks unsatisfying imho.

i don't think it's unsatisfying, you would simply return from within the brackets.